source/CCustomOpcodeSystem.cpp

```diff
@@ -1782,33 +1713,36 @@ namespace CLEO
 	//0ADE=2,%2d% = text_by_GXT_entry %1d%
 	OpcodeResult __stdcall opcode_0ADE(CRunningScript *thread)
 	{
-		auto gxt = ReadStringParam(thread); OPCODE_VALIDATE_STR_ARG_READ(gxt)
+		auto gxt = OPCODE_READ_PARAM_STRING();
 
-		if (*thread->GetBytePointer() >= 1 && *thread->GetBytePointer() <= 8)
-			*thread << GetInstance().TextManager.Get(gxt);
+		auto txt = GetInstance().TextManager.Get(gxt);
+
+		if (IsVarString(thread->PeekDataType()))
+		{
+			OPCODE_WRITE_PARAM_STRING(txt);
+		}
 		else
 		{
-			auto ok = WriteStringParam(thread, GetInstance().TextManager.Get(gxt)); OPCODE_VALIDATE_STR_ARG_WRITE(ok)
+			OPCODE_WRITE_PARAM_PTR(txt); // address of the text
 		}
-
 		return OR_CONTINUE;
 	}
 
 	//0ADF=2,add_dynamic_GXT_entry %1d% text %2d%
 	OpcodeResult __stdcall opcode_0ADF(CRunningScript *thread)
 	{
-		char gxtLabel[8]; // 7 + terminator character
-		auto gxtOk = ReadStringParam(thread, gxtLabel, sizeof(gxtLabel)); OPCODE_VALIDATE_STR_ARG_READ(gxtOk)
-		auto text = ReadStringParam(thread); OPCODE_VALIDATE_STR_ARG_READ(text)
+		char gxtLabel[9] = { 0 }; // 8 + terminator character
+		auto gxt = OPCODE_READ_PARAM_STRING_BUFF(gxtLabel, 8);

```

Comment:
why do we increase the length here?

Reply:
So temporary buffer is able to hold whole GXT labels as cstr.

Reply:
It's already limited to 7 characters + null terminator (always has been). Why do we change it now?

Reply:
not sure which fix you mean? can you point out?

short string variables can only be used with fixed-length string literals (represented with single quotes in SB), they occupy 8 bytes: 7 characters and null-terminators. Any value in a short-string variable will be also null-terminated. If not, then this is a mistake.

Reply:
I was fixing shortString and longString variables support, as they do not need terminator. There is no reason for null termination if you know fixed max length of carrier type.

Reply:
GXT labels are always 7 characters max. There is no need for this change

Reply:
Oh ok. While short string can hold up to 8 chars I was not able to find single GXT label longer than 7 chars.

Reply:
Fixed/